### PR TITLE
fix: Load proc-macros for rustc_private crates

### PR DESCRIPTION
**Pull request by Veykril**
Fixes https://github.com/rust-lang/rust-analyzer/issues/13591

Verified that this makes things work in the clippy repo (like resolving `sym` things for example)

### CODE REVIEW

File: crates/project-model/src/build_scripts.rs [hunk 1]
```diff
@@ -378,6 +378,80 @@ impl WorkspaceBuildScripts {
...
+            tracing::info!("Loading rustc proc-macro paths from {}", target_libdir.display());
+
+            let proc_macro_dylibs: Vec<(String, AbsPathBuf)> = std::fs::read_dir(target_libdir)?
+                .filter_map(|entry| {
+                    let dir_entry = entry.ok()?;
+                    if dir_entry.file_type().ok()?.is_file() {
+                        let path = dir_entry.path();
+                        tracing::info!("p{:?}", path);
+                        let extension = path.extension()?;
+                        if extension == "dll" || extension == "so" {
```
Comment (bjorn3) [inline crates/project-model/src/build_scripts.rs:419]: macOS uses .dylib as extension. I did recommend using `std::env::consts::DLL_EXTENSION` instead. We can't load proc macros for another target anyway.

File: crates/project-model/src/build_scripts.rs [hunk 2]
```diff
@@ -378,6 +378,80 @@ impl WorkspaceBuildScripts {
...
+                            let name = path.file_stem()?.to_str()?.split_once('-')?.0.to_owned();
+                            let path = AbsPathBuf::try_from(path).ok()?;
+                            return Some((name, path));
+                        }
+                    }
+                    None
+                })
+                .collect();
+            for p in rustc.packages() {
```
Comment (bjorn3) [inline crates/project-model/src/build_scripts.rs:428]: Shouldn't this check that the package is actually a proc macro? It seems like the current logic will attempt to load rustc_driver as proc macro as it is a dylib too.

File: crates/project-model/src/build_scripts.rs (hunk 1 above)
Comment (Veykril) [reply crates/project-model/src/build_scripts.rs:419]: Ah, that's what I forgot to do properly 👍

File: crates/project-model/src/build_scripts.rs (hunk 2 above)
Comment (Veykril) [reply crates/project-model/src/build_scripts.rs:428]: It certainly should, I assumed there would only be proc-macro dylibs in there but that's clearly not a good assumption (and even then the check is better)

**Pull request merged**